lib/dotcom_web/components/live_components/trip_planner_results_section.ex

```diff
@@ -16,27 +16,56 @@ defmodule DotcomWeb.Components.LiveComponents.TripPlannerResultsSection do
 
   @impl true
   def render(assigns) do
+    grid_column_layout =
+      case assigns.results.result do
+        nil -> "md:grid-cols-[1fr]"
+        _ -> "md:grid-cols-[1fr_1fr]"
+      end
+
+    assigns = assign(assigns, :grid_column_layout, grid_column_layout)
+
     ~H"""
-    <section class="flex w-full border border-solid border-slate-400">
+    <section class={"flex flex-col md:grid #{@grid_column_layout} md:grid-rows-[min-content_1fr] w-full border border-solid border-slate-400"}>
       <div :if={@error} class="w-full p-4 text-rose-400">
         <%= inspect(@error) %>
       </div>
       <.async_result :let={results} assign={@results}>
-        <div :if={results} class="w-full p-4">
+        <div
+          :if={results && @expanded_itinerary_index}
+          class="row-start-1 col-start-1 h-min w-full p-4"
+        >
+          <button
+            type="button"
+            phx-click="set_expanded_itinerary_index"
+            phx-value-index="nil"
+            phx-target={@myself}
+            class="btn-link"
+          >
+            <span class="flex flex-row items-center">
+              <.icon class="fill-brand-primary h-4 mr-2" name="chevron-left" />
+              <span class="font-medium">View All Options</span>
+            </span>
+          </button>
+        </div>
+      </.async_result>
+
+      <.map
+        map_config={@map_config}
+        from={@from}
+        to={@to}
+        hide_on_mobile={@expanded_itinerary_index == nil}
+        class="row-span-2"
+      />
+
+      <.async_result :let={results} assign={@results}>
```

Comment:
Something about how I re-arranged the DOM elements added some extra spacing between `View all options` and the content below it.

I'm currently leaning towards leaving it as-is for this PR, and fixing it, as well as a bunch of other minor visual issues as a follow-up.